src/libs/kata-types/src/config/hypervisor/dragonball.rs

```diff
@@ -0,0 +1,185 @@
+            }
+
+            if db.memory_info.default_memory == 0 {
+                db.memory_info.default_memory = default::DEFAULT_DB_MEMORY_SIZE;
+            }
+            if db.memory_info.memory_slots == 0 {
+                db.memory_info.memory_slots = default::DEFAULT_DB_MEMORY_SLOTS;
+            }
+        }
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    fn validate(&self, conf: &TomlConfig) -> Result<()> {
+        if let Some(db) = conf.hypervisor.get(HYPERVISOR_NAME_DRAGONBALL) {
+            if !db.path.is_empty() {
+                return Err(eother!("Path for dragonball hypervisor should be empty"));
+            }
+            if !db.valid_hypervisor_paths.is_empty() {
+                return Err(eother!(
+                    "Valid hypervisor path for dragonball hypervisor should be empty"
+                ));
+            }
+            if !db.ctlpath.is_empty() {
+                return Err(eother!("CtlPath for dragonball hypervisor should be empty"));
+            }
+            if !db.valid_ctlpaths.is_empty() {
+                return Err(eother!("CtlPath for dragonball hypervisor should be empty"));
+            }
+            validate_path!(db.jailer_path, "Dragonball jailer path {} is invalid: {}")?;
+            if db.enable_iothreads {
+                return Err(eother!("Dragonball hypervisor doesn't support IO threads."));
+            }
+
+            if !db.blockdev_info.disable_block_device_use
+                && db.blockdev_info.block_device_driver != VIRTIO_BLK
+                && db.blockdev_info.block_device_driver != VIRTIO_BLK_MMIO
+                && db.blockdev_info.block_device_driver != VIRTIO_PMEM
+            {
+                return Err(eother!(
+                    "{} is unsupported block device type.",
+                    db.blockdev_info.block_device_driver
+                ));
+            }
+
+            if db.boot_info.kernel.is_empty() {
+                return Err(eother!(
+                    "Guest kernel image for dragonball hypervisor is empty"
+                ));
+            }
+            if db.boot_info.image.is_empty() {
+                return Err(eother!(
+                    "Guest boot image for dragonball hypervisor is empty"
+                ));
+            }
+            if !db.boot_info.initrd.is_empty() {
+                return Err(eother!("Initrd for dragonball hypervisor should be empty"));
+            }
+            if !db.boot_info.firmware.is_empty() {
+                return Err(eother!(
+                    "Firmware for dragonball hypervisor should be empty"
+                ));
+            }
+
+            if (db.cpu_info.default_vcpus > 0
+                && db.cpu_info.default_vcpus as u32 > default::MAX_DB_VCPUS)
+                || db.cpu_info.default_maxvcpus > default::MAX_DB_VCPUS
+            {
+                return Err(eother!(
+                    "Dragonball hypervisor can not support {} vCPUs",
```

Comment:
**validate** will be used after loading the toml file into the

Reply:
the execution order is to load `Toml` file to TomlConfig first, then `validate` Tomlconfig. The function is not being called is because, we depend on the default config to do the testing for updating annotation. However, path. ctlpath and jailer path are